all-of-us/workbench
# [RW-11078][risk=no]  Backfill Absorb External Department ID for existing users

Please note: This script will run just once for all users per env. 

**Purpose of this script is to add the following information to all the existing users:**
          if (configProvider.get().absorb.externalDepartmentIdPopulatedForNewUsers) {
                // The value of this field must match one of the manually-configured values in the
                // Absorb installation.
                // See https://docs.google.com/document/d/1xgcvow0xPL6K4vxyM3PVlW-1E5Ye3X1Y5wZSD5rrPlc
                aouCustomFields.put(
                    GSUITE_FIELD_ABSORB_EXTERNAL_DEPARTMENT_ID,
                    configProvider.get().absorb.externalDepartmentId);
              }
 
**How to run**

Dry-run:  ./project.rb backfill-absorb  --project all-of-us-workbench-test --dry-run true

w/o dry run:  ./project.rb backfill-absorb  --project all-of-us-workbench-test  or  ./project.rb backfill-absorb  --project all-of-us-workbench-test --dry-run false

**How to test:**
1) Existing user will not be able to access absrob certificates/lessons
2) Run script
3) Existing user will be able to see absrob certificates/lessons


---
**PR checklist**

- One or more of the following is true:
